AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/States/AuthorizationState.swift

```diff
@@ -19,8 +19,10 @@ enum AuthorizationState: State {
 
     case clearingFederation
 
-    case federatingToIdentityPool(FetchAuthSessionState,
-                                  FederatedToken)
+    case federatingToIdentityPool(
+        FetchAuthSessionState,
+        FederatedToken,
+        existingCredentials: AmplifyCredentials)
```

Comment:
Why was this `existingCredentials` added? Not sure where this is used

Reply:
Had a discussion offline..

> So this was added because when session error happens during federation, it expects to have a credential passed to it, so that fetchAuthSession can determine the type of retry that could happen.. case sessionError(FetchSessionError, AmplifyCredentials)